tests/integration/targets/elb_application_lb/tasks/main.yml

```diff
@@ -1,12 +1,1172 @@
+        state: present
+        tags: {}
+      register: elb
+
+    - assert:
+        that:
+          - elb is changed
+          - not elb.tags
+
+    - name: Update an ELB by removing tags (idempotence) - check_mode
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group.group_id }}"
+        state: present
+        tags: {}
+      register: elb
+      check_mode: yes
+
+    - assert:
+        that:
+          - elb is not changed
+          - elb.msg is match('IN CHECK MODE - no changes to make to ELB specified.')
+
+    - name: Update an ELB by removing tags (idempotence)
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group.group_id }}"
+        state: present
+        tags: {}
+      register: elb
+
+    - assert:
+        that:
+          - elb is not changed
+          - not elb.tags
+
+    # ------------------------------------------------------------------------------------------
+
+    - name: Update an ELB by changing security group - check_mode
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group2.group_id }}"
+        state: present
+      register: elb
+      check_mode: yes
+
+    - assert:
+        that:
+          - elb is changed
+          - elb.msg is match('Would have updated ELB if not in check mode.')
+
+    - name: Update an ELB by changing security group
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group2.group_id }}"
+        state: present
+      register: elb
+
+    - assert:
+        that:
+          - elb is changed
+          - elb.security_groups[0] == sec_group2.group_id
+
+    - name: Update an ELB by changing security group (idempotence) - check_mode
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group2.group_id }}"
+        state: present
+      register: elb
+      check_mode: yes
+
+    - assert:
+        that:
+          - elb is not changed
+          - elb.msg is match('IN CHECK MODE - no changes to make to ELB specified.')
+
+    - name: Update an ELB by changing security group (idempotence)
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group2.group_id }}"
+        state: present
+      register: elb
+
+    - assert:
+        that:
+          - elb is not changed
+          - elb.security_groups[0] == sec_group2.group_id
+
+    # ------------------------------------------------------------------------------------------
+
+    - name: Ensure elb_application_lb_info supports check_mode
+      elb_application_lb_info:
+      register: elb_info
+      check_mode: yes
```

Comment:
Not sure what others think but I didn't think check mode was really applicable for info modules as nothing really changes

Reply:
I know that all the modules should support check_mode. You're right, there is no change. _info modules are a particular case, but don't know about the story behind the check_mode here.

Reply:
Yeah makes sense, guess they are indeed quite a particular case

Reply:
Yeah, I was basing that off a conversation with @jillr who basically said it should be there so nothing breaks if the user does --check on an entire playbook

Reply:
Ahhh cool cool 👍🏻